python/cugraph/cugraph/dask/structure/mg_property_graph.py

```diff
@@ -411,22 +443,20 @@ def add_edge_data(self,
         # columns. The copied DataFrame is then merged (another copy) and then
         # deleted when out-of-scope.
         tmp_df = dataframe.copy()
-        # FIXME: Find a better way to create the edge id
-        prev_eid = -1 if self.__last_edge_id is None else self.__last_edge_id
         tmp_df[self.src_col_name] = tmp_df[vertex_col_names[0]]
         tmp_df[self.dst_col_name] = tmp_df[vertex_col_names[1]]
+        tmp_df[self.type_col_name] = type_name
+
+        # Add unique edge IDs to the new rows
+        prev_eid = -1 if self.__last_edge_id is None else self.__last_edge_id
         starting_eid = prev_eid + 1
         data_size = len(tmp_df.compute().index)
-        cudf_series = \
-            cudf.Series(range(starting_eid, starting_eid + data_size))
-        dask_series =\
-            dask_cudf.from_cudf(cudf_series, self.__num_workers)
-        dask_series = dask_series.reset_index(drop=True)
-        self.__last_edge_id = starting_eid + data_size
-        tmp_df = tmp_df.reset_index(drop=True)
-        tmp_df[self.edge_id_col_name] = dask_series
-        tmp_df[self.type_col_name] = type_name
+        new_eids = cudf.Series(range(starting_eid, starting_eid + data_size))
+        new_eids_dask = dask_cudf.from_cudf(new_eids, self.__num_workers)
+        self.__last_edge_id = starting_eid + data_size - 1
+        tmp_df[self.edge_id_col_name] = new_eids_dask
```

Comment:
I think we can just do this via below API and avoid the `cuDF` and client calls and also keeps it delayed. 

```python
tmp_df[self.edge_id_col_name] = 1
tmp_df[self.edge_id_col_name] = tmp_df[self.edge_id_col_name]].cumsum()+starting_eid 

self.__last_edge_id = starting_eid + data_size - 1
```

Reply:
This is much better, thanks. I had to change it slightly to keep the edge IDs starting at 0, but all tests are passing. I also used this same pattern for SG to better accommodate a refactoring where SG and MG are sharing the core implementation more (cc @eriknw) and SG tests are also passing.